src/storage/file_storage.cpp

```diff
@@ -65,78 +65,78 @@ ErrorCode FileStorage::create( const std::string& path, const FileStorageConfig&
   m_configFile.seekp(0,std::ios_base::beg);
   m_configFile.write(m_pageFiller.data(), m_pageFiller.size());
   if(!m_configFile) {
-    return ErrorCode::E_STORAGE_OUT_OF_BOUNDS_WRITE;
+    throw ErrorCode::E_STORAGE_OUT_OF_BOUNDS_WRITE;
   }
 
   // Write FileStorageConfig in the first page of m_configFile
   m_configFile.seekp(0,std::ios_base::beg);
   m_configFile.write(reinterpret_cast<char*>(&m_config), sizeof(m_config));
   m_configFile.flush();
   if(!m_configFile) {
-    return ErrorCode::E_STORAGE_OUT_OF_BOUNDS_WRITE;
+    throw ErrorCode::E_STORAGE_OUT_OF_BOUNDS_WRITE;
   }
   return ErrorCode::E_NO_ERROR;
 }
 
-ErrorCode FileStorage::close() noexcept {
+ErrorCode FileStorage::close() {
   if(m_dataFile) {
     m_dataFile.close();
   }
   else {
-    return ErrorCode::E_STORAGE_NOT_OPEN;
+    throw ErrorCode::E_STORAGE_NOT_OPEN;
   }
 
   if(m_configFile) {
     m_configFile.close();
   }
   else {
-    return ErrorCode::E_STORAGE_NOT_OPEN;
+    throw ErrorCode::E_STORAGE_NOT_OPEN;
   }
 
   return ErrorCode::E_NO_ERROR;
 }
 
-ErrorCode FileStorage::reserve( const uint32_t& numPages, pageId_t* pageId ) noexcept {
+ErrorCode FileStorage::reserve( const uint32_t& numPages, pageId_t* pageId ) {
   m_dataFile.seekp(0,std::ios_base::end);
   if(!m_dataFile) {
-    return ErrorCode::E_STORAGE_CRITICAL_ERROR;
+    throw ErrorCode::E_STORAGE_CRITICAL_ERROR;
   }
   *pageId = bytesToPage(m_dataFile.tellp());
   m_dataFile.seekp(pageToBytes((numPages-1)),std::ios_base::end);
   m_dataFile.write(m_pageFiller.data(), m_pageFiller.size());
   if(!m_dataFile) {
```

Comment:
I would not throw this error as an out of bounds error, since an error in a write operation can be caused by many things (e.g. out of space for instance). Thus, I would throw something more generic such "E_UNEXPECTED_WRITE_ERROR" or something like this

Reply:
Solved in 537a316.